src/services/TestDataService.ts

```diff
@@ -1185,6 +1234,26 @@ export class TestDataService {
         return salesChannel;
     }
 
+    /**
+     * Assigns a language to a sales channel.
+     *
+     * @param salesChannelId - The uuid of the sales channel.
+     * @param languageId - The uuid of the language.
+     */
+    async assignSalesChannelLanguage(salesChannelId: string, languageId: string) {
```

Comment:
I would go here not with a patch, because then we could not make it back and stay the whole test execution with the assigned language within the Sales Channel. That would affect other test cases. I would go here with a sync API command for the table sales-channel-language...here a snippet. 

const syncSalesChannelResponse = await this.AdminApiClient.post('./_action/sync', {
            data: {
                'write-sales-channel-language': {
                    entity: 'sales_channel_language',
                    action: 'upsert',
                    payload: [
                        {
                            salesChannelId: salesChannelId,
                            languageId: languageId,
                        },
                    ],
                },
            },
        });
        expect(syncSalesChannelResponse.ok()).toBeTruthy();

        const { data: salesChannel } = await syncSalesChannelResponse.json();

        this.addCreatedRecord('sales_channel_language', {salesChannelId: salesChannelId, languageId: languageId})

    private highPriorityEntities = ['order', 'product', 'landing_page', 'sales_channel_currency', 'sales_channel_country', 'sales_channel_language'];

Reply:
Updated. I need to delete customer before the **sales_channel_language**, if not the customers those linked to the added language won't be deleted.

Reply:
mhh i will double check

Reply:
Okay...I double check it and your current implementation is doing what it should. I assign a language to the sales channel (available in the sales channel settings) and also delete it properly. But if you add a custom language and assign it to your sales channel you have to add to the priority list: `sales_channel_language` ...because that entry has to be deleted first before the `language` entity is deleted

Reply:
Regarding your problem with the customer deletion it is another one and not related to that thread.